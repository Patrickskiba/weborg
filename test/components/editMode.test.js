import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'
import userEvent from 'user-event'
import 'jest-dom/extend-expect'

describe('editMode tests', () => {
    afterEach(cleanup)

    it('renders 3 editiable fields and can save them', () => {
        const mockProps = {
            editNode: JSON.parse("{\"type\":\"headline\",\"index\":1,\"level\":2,\"content\":[{\"type\":\"text\",\"text\":\"rsync\"}],\"children\":[{\"type\":\"section\",\"index\":2,\"content\":[{\"type\":\"text\",\"text\":\" Copy a file with a progress bar\"}]},{\"type\":\"section\",\"index\":3,\"content\":[{\"type\":\"text\",\"text\":\" sudo rsync --info=progress2 source dest\"}]}]}"),
            setEditNode: jest.fn(),
            text: `* Great Unix Tools\n** rsync\nCopy a file with a progress bar\nsudo rsync --info=progress2 source dest\n** du - disk usage\ndu -sh file_path\n-s : summarized\n-h : human readable\n** pacman\nsearch pacman\n- sudo pacman -Ss package_name`,
            setText: jest.fn()
        }

        const EditMode = require('../../src/components/EditMode').default
        const { getByText, getByLabelText, container } = render(<EditMode {...mockProps} />)

        
        const level = getByLabelText('Level')
        const headline = getByLabelText('Headline')
        const content = getByLabelText('Content')

        expect(level.value).toEqual('2')

        expect(headline.value).toEqual('rsync')

        expect(content.value).toEqual(' Copy a file with a progress bar\n sudo rsync --info=progress2 source dest')

        expect(container).toMatchSnapshot()
    })

    it('clicking save button does not change original text when making no change', () => {
        const mockProps = {
            editNode: JSON.parse("{\"type\":\"headline\",\"index\":1,\"level\":2,\"content\":[{\"type\":\"text\",\"text\":\"rsync\"}],\"children\":[{\"type\":\"section\",\"index\":2,\"content\":[{\"type\":\"text\",\"text\":\" Copy a file with a progress bar\"}]},{\"type\":\"section\",\"index\":3,\"content\":[{\"type\":\"text\",\"text\":\" sudo rsync --info=progress2 source dest\"}]}]}"),
            setEditNode: jest.fn(),
            text: `* Great Unix Tools\n** rsync\n Copy a file with a progress bar\n sudo rsync --info=progress2 source dest\n** du - disk usage\ndu -sh file_path\n-s : summarized\n-h : human readable\n** pacman\nsearch pacman\n- sudo pacman -Ss package_name`,
            setText: jest.fn()
        }

        const EditMode = require('../../src/components/EditMode').default
        const { getByText, getByLabelText, container } = render(<EditMode {...mockProps} />)

        const save = getByText('Click me')

        fireEvent.click(save, { button: 1 })

        expect(mockProps.setText).toHaveBeenCalledWith(mockProps.text)
    })

    it('clicking save button when making a change will set the correct text', () => {
        const mockProps = {
            editNode: JSON.parse("{\"type\":\"headline\",\"index\":1,\"level\":2,\"content\":[{\"type\":\"text\",\"text\":\"rsync\"}],\"children\":[{\"type\":\"section\",\"index\":2,\"content\":[{\"type\":\"text\",\"text\":\" Copy a file with a progress bar\"}]},{\"type\":\"section\",\"index\":3,\"content\":[{\"type\":\"text\",\"text\":\" sudo rsync --info=progress2 source dest\"}]}]}"),
            setEditNode: jest.fn(),
            text: `* Great Unix Tools\n** rsync\n Copy a file with a progress bar\n sudo rsync --info=progress2 source dest\n** du - disk usage\ndu -sh file_path\n-s : summarized\n-h : human readable\n** pacman\nsearch pacman\n- sudo pacman -Ss package_name`,
            setText: jest.fn()
        }

        const EditMode = require('../../src/components/EditMode').default
        const { getByText, getByLabelText, container } = render(<EditMode {...mockProps} />)

        const level = getByLabelText('Level')
        userEvent.type(level, '3')

        const headline = getByLabelText('Headline')
        userEvent.type(headline, 'new headline')

        const content = getByLabelText('Content')
        userEvent.type(content, 'new content\nwith new line')

        const save = getByText('Click me')

        fireEvent.click(save, { button: 1 })

        expect(mockProps.setText).toHaveBeenCalledWith(`* Great Unix Tools\n*** new headline\nnew content\nwith new line\n** du - disk usage\ndu -sh file_path\n-s : summarized\n-h : human readable\n** pacman\nsearch pacman\n- sudo pacman -Ss package_name`)
    })
})
