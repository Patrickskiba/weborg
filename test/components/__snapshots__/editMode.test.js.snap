// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editMode tests renders 3 editiable fields and can save them 1`] = `
<div>
  <div>
    <button>
      Clickith me
    </button>
    <div>
      <label
        for="headline-level-input"
      >
        Level
      </label>
      <input
        class="sc-bdVaJa gZdsYr"
        id="headline-level-input"
        type="text"
        value="2"
      />
    </div>
    <div>
      <label
        for="headline-text-input"
      >
        Headline
      </label>
      <input
        class="sc-bwzfXH boSKPF"
        id="headline-text-input"
        type="text"
        value="rsync"
      />
    </div>
    <div>
      <label
        for="section-text-input"
      >
        Content
      </label>
      <textarea
        class="sc-htpNat gDPqZm"
        id="section-text-input"
      >
         Copy a file with a progress bar
 sudo rsync --info=progress2 source dest
      </textarea>
    </div>
    <div>
      <button>
        Click me
      </button>
    </div>
  </div>
</div>
`;
